File: 2788. 按分隔符拆分字符串.cpp
/*
 * @lc app=leetcode.cn id=2788 lang=cpp
 *
 * [2788] 按分隔符拆分字符串
 */

// @lc code=start
class Solution {
public:
    vector<string> splitWordsBySeparator(vector<string>& words, char separator) {
        vector<string> res;
        for(auto& word : words){
            int idx = 0;
            for(int i = 0;i<=word.size();++i){
                if(word[i] == separator || i == word.size()){
                    if(idx != i)res.emplace_back(word.substr(idx, i - idx));
                    idx = i + 1;
                }
            }
        }
        return res;
    }
};
// @lc code=end

File: 2788. 按分隔符拆分字符串.rs
/*
 * @lc app=leetcode.cn id=2788 lang=rust
 *
 * [2788] 按分隔符拆分字符串
 */

// @lc code=start
impl Solution {
    pub fn split_words_by_separator(words: Vec<String>, separator: char) -> Vec<String> {
        words.iter().flat_map(|s| s.split(separator)).map(|s| s.to_string()).filter(|s| !s.is_empty()).collect::<Vec<_>>()
    }
}
// @lc code=end